### PR TITLE
fix: De-dupe current version history entries and address some minor typos

### PR DESCRIPTION
**Pull request by agrawroh**
While working on a feature and adding release notes for it, I noticed that there is a duplicate entry and also some minor typos here and there. This is a small clean-up PR to remove the duplicate entry and correct the typos.

**Commit Message:** de-dupe current version history entries and address some minor typos.
**Additional Description:** De-dupes the duplicate entires in the current version's history file and also address some minor typos/issues.
**Risk Level:** None
**Testing:** N/A
**Docs Changes:** De-duped the duplicate entries and fixed typos etc.
**Release Notes:** Fixed
**Platform Specific Features:** N/A

**Signed-off-by:** Rohit Agrawal <rohit.agrawal@databricks.com>

### CODE REVIEW

**Comment by agrawroh** [comment]
@snowp @dio There are very small cosmetic changes in the version history file. It shouldn't take more than 5-10 mins to review. Could you please take a look whenever you get some free time?

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 Incompatible Behavior Changes
 -----------------------------
-*Changes that are expected to cause an incompatibility if applicable; deployment changes are likely required*
+*Changes that are expected to cause incompatibility if applicable; deployment changes are likely required*
```
Comment (phlax) [inline docs/root/version_history/current.rst:6]: i think either way works here

File: docs/root/version_history/current.rst [hunk 2]
```diff
@@ -82,10 +81,11 @@ Bug Fixes
 * ext_authz: fix the HTTP ext_authz filter to respond with ``403 Forbidden`` when a gRPC auth server sends a denied check response with an empty HTTP status code.
 * ext_authz: the network ext_authz filter now correctly sets dynamic metadata returned by the authorization service for non-OK responses. This behavior now matches the http ext_authz filter.
 * hcm: remove deprecation for :ref:`xff_num_trusted_hops <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.xff_num_trusted_hops>` and forbid mixing ip detection extensions with old related knobs.
-* http: limit use of deferred resets in the http2 codec to server-side connections. Use of deferred reset for client connections can result in incorrect behavior and performance problems.
+* http: limit use of deferred resets in the http2 codec to server-side connections. The use of deferred reset for client connections can result in incorrect behavior and performance problems.
```
Comment (phlax) [inline docs/root/version_history/current.rst:84]: this probs works either way

File: docs/root/version_history/current.rst [hunk 3]
```diff
@@ -136,4 +136,4 @@ Deprecated
   :ref:`contains_match <envoy_v3_api_field_config.route.v3.HeaderMatcher.contains_match>` are deprecated by :ref:`string_match <envoy_v3_api_field_config.route.v3.HeaderMatcher.string_match>`.
 * listener: :ref:`reuse_port <envoy_v3_api_field_config.listener.v3.Listener.reuse_port>` has been
   deprecated in favor of :ref:`enable_reuse_port <envoy_v3_api_field_config.listener.v3.Listener.enable_reuse_port>`.
-  At the same time, the default has been changed from false to true. See above for more information.
+  At the same time, the default is getting changed from false to true. See above for more information.
```
Comment (phlax) [inline docs/root/version_history/current.rst:139]: i think the original is probably more correct here

**Review by phlax — COMMENTED** [review]
hi @agrawroh thanks for this, as discussed offline there are a few changes im ambivalent about, but otherwise looks good

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (agrawroh) [reply docs/root/version_history/current.rst:6]: Reverted.

File: docs/root/version_history/current.rst (hunk 2 above)
Comment (agrawroh) [reply docs/root/version_history/current.rst:84]: Reverted.

File: docs/root/version_history/current.rst (hunk 3 above)
Comment (agrawroh) [reply docs/root/version_history/current.rst:139]: Reverted.

File: docs/root/version_history/current.rst [hunk 4]
```diff
@@ -86,6 +85,7 @@ Bug Fixes
 * listener: fixed an issue on Windows where connections are not handled by all worker threads.
 * lua: fix ``BodyBuffer`` setting a Lua string and printing Lua string containing hex characters. Previously, ``BodyBuffer`` setting a Lua string or printing strings with hex characters will be truncated.
 * xray: fix the AWS X-Ray tracer bug where span's error, fault and throttle information was not reported properly as per the `AWS X-Ray documentation <https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html>`_. Before this fix, server error was reported under 'annotations' section of the segment data.
+* xray: fix the AWS X-Ray tracer bug where span's error, fault and throttle information was not reported properly as per the `AWS X-Ray documentation <https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html>`_. Before this fix, server error was reported under the 'annotations' section of the segment data.
```
Comment (phlax) [inline docs/root/version_history/current.rst:88]: i think this adds a dupe entry

**Review by phlax — CHANGES_REQUESTED** [review]
hi @agrawroh 

i think there is one dupe added apart from the cleanups, if we can remove that im happy to land

File: docs/root/version_history/current.rst (hunk 4 above)
Comment (agrawroh) [reply docs/root/version_history/current.rst:87]: Done, Thanks!

**Review by phlax — APPROVED** [review]
lgtm, thanks @agrawroh

**Comment by phlax** [comment]
@agrawroh as a general point its generally best not to force-push (unless really required) once you have reviewer/s for a PR

it make the reviewers job more time consuming as they have to check all changes again

**Comment by agrawroh** [comment]
> @agrawroh as a general point its generally best not to force-push (unless really required) once you have reviewer/s for a PR
> 
> it make the reviewers job more time consuming as they have to check all changes again

Make sense. I mistakenly added `--rebase` while pulling the latest master and hence had to force push. I'll be careful from next time.

**Comment by agrawroh** [comment]
@phlax Looks like I don’t have merge permissions :(

**Pull request merged**